### DISCUSSION TITLE
question about pagination with collection

### DISCUSSION THREAD

**Discussion by btrem** [11ty/eleventy#2524] [Q&A] [answered]
I'm using a collection of posts by year using this snippet in .eleventy.js:

        eleventyConfig.addCollection('postsByYear', function(collectionApi) {

            let postDate, yyyy;
            let postsByYear = {};
            let posts = collectionApi.getFilteredByGlob('src/posts/**/*.*');

            posts.forEach(function(post) {
                postDate = new Date(post.data.date);
                yyyy = postDate.getFullYear().toString();

                // create a new array key with the year if it doesn't already exist
                if (!postsByYear[yyyy]) {
                    postsByYear[yyyy] = [];
                }
                // add post to the year array key
                postsByYear[yyyy].push(post);
            });
            return postsByYear;
        });

Then I have this template in the root of my src directory:

    ---
    pagination:
        data: collections.postsByYear
        size: 1
        alias: year
    permalink: /{{ year }}/
    ---
    {% for post in collections.postsByYear[year] | reverse %}
    
    {%- endfor %}

It creates an index of posts by year, e.g. `/2021/index.html` and `/2022/index.html`. But there are aspects that I don't understand. For instance, what does the alias year refer to? The data is a collection. Is year the index of an array of posts?

I ask because I am having nothing but trouble trying to extend this to months, e.g. `/2022/07/index.html`.

**Comment by pdehaan** [comment, marked as answer]
> _"But there are aspects that I don't understand. For instance, what does the alias year refer to? The data is a collection. Is year the index of an array of posts?"_

It looks like your `collections.postsByYear` collection returns an Object (where the stringified `year` is the key, and array of posts is the value). See https://www.11ty.dev/docs/pagination/#paging-an-object
> _"Eleventy does allow paging objects too. Objects are resolved to pagination arrays using either the `Object.keys()` or `Object.values()` JavaScript functions."_

So in your case, I _think_ the `year` alias is one of the keys of the `collections.postsByYear` object.

Not sure the best way to extend this to year+month, since that'd be some non-trivial nested double pagination.
My first thought would be maybe to copy/paste your code and create a second collection named `collections.postsByYearMonth` and then set the keys to something like `2022_08` (or `2022/08`) with an array value of posts in that year+month combination, and then have to tweak the `permalink` to get the output file structure that you want.
